### ISSUE TITLE
[UI/Mobile] Idea of mobile version

### ISSUE THREAD

**Issue by LuigiElleBalotta** [Reinisch/Warcraft-Arena-Unity#11]
I don't know unity very well, but I've an idea to play on mobile.
We should check if we are compiling for desktop or Android/iOS.
We already have everything for desktop.
On mobile we should draw the GUI in a different way.

The camera can be rotated using the finger's swipe. 
There should be a virtual joystick on the left to move the player (we can have the same camera script).
Some spell on the right side of the screen. We can re-use already implemented spell with an addiction:
Spell should have an Enum that is structed as:
- Mobile Only
- Desktop Only
- Both

This enum tells the server how many spell i can use (and show) on mobile and/or desktop mode.

The target:
We can choose to auto-aim the target as lots of warcraft-like games, we can draw a pointer in the center of the screen and enable spell casting when we have a target in the pointer (so we aim the target, start spell casting and if the target is moving the camera will follow the target).
OR we can use warcraft similar system, so we have to touch the target then we can cast.

For mobile i think chat should be removed ( CoD mobile uses vocal chat, but i think it's not needed ).

For player/target health/mana GUI object there are 2 ways:
draw a simple bar over our head or resize the existing GUI using some responsive techniques

 I hope this is an interesting idea :)

**Comment by Reinisch** [comment]
Android and iOS should definitely compile, the only plugin used is Photon and it works on mobile.

I don't think it is good to limit casting on different platforms, instead spells may be casted by using multiple buttons, for example first press will open fire school category, and that will open another menu with fire spells. In addition the most used spells will have their buttons separately.

**Comment by LuigiElleBalotta** [comment]
I didnt' said it won't compile for android/iOS, but with this ui the game will be unplayable
By the way, your solution is good too, but is it fast enough to play? I imagine the healer, it should click on the container and on the healing spell every time?

**Comment by Reinisch** [comment]
Yeah may be a little too slow, not sure how to do it otherwise, I need to finish some missing logic before starting ui anyway.

**Comment by LuigiElleBalotta** [comment]
Maybe flagging a spell as MobileOnly/Both/DesktopOnly it's faster.
You choose only N spells to assign to the android player. And you can change those spell only in certain conditions (example: i'm horde player and i'm inside orgrimmar map/dalaran map/map lobby). 
If i know i'm the healer i will choose only healing spells and maybe 1 damage spell and 1 buff.. 
I think this has sense :D
